src/redux/books/booksSlice.js

```diff
@@ -1,47 +1,59 @@
-const ADD_BOOK = 'Bookstore/books/ADD_BOOK';
-const REMOVE_BOOK = 'Bookstore/books/REMOVE_BOOK';
-
-const initialState = [
-  {
-    id: '1',
-    title: 'The Great Gatsby',
-    author: 'John Smith',
-  },
-  {
-    id: '2',
-    title: 'Anna Karenina',
-    author: 'Leo Tolstoy',
-  },
-  {
-    id: '3',
-    title: 'The Selfish Gene',
-    author: 'Richard Dawkins',
-  },
-];
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import BookSrvc from '../booksAPI';
 
+const initialState = {
+  books: [],
+};
 // Actions
 
-export const addBook = (book) => ({
-  type: ADD_BOOK,
-  payload: book,
-});
+const fetchBooks = createAsyncThunk(
+  'books/fetchBooks',
+  async () => {
+    const { data } = await BookSrvc.getSrvc();
+    return Object.keys(data).map((key) => ({ ...data[key][0], item_id: key }));
+  },
+);
 
-export const removeBook = (id) => ({
-  type: REMOVE_BOOK,
-  payload: id,
-});
+const addBook = createAsyncThunk(
+  'books/addBook',
+  async (Book) => {
+    const response = await BookSrvc.addSrvc(Book);
+    return response.data;
+  },
+);
+
+const deleteBook = createAsyncThunk(
+  'books/deleteBook',
+  async (id) => {
+    await BookSrvc.deleteSrvc(id);
+    return id;
+  },
+);
 
 // Reducer
 
 const bookReducer = (state = initialState, action) => {
```

Comment:
- [ ] I also want to kindly suggest you make use of redux toolkit `CreateSlice` to create your reducer and actions,